static/js/index.js

```diff
@@ -970,6 +1007,61 @@ ep_comments.prototype.saveComment = function(data, rep) {
   });
 }
 
+ep_comments.prototype.saveCommentWithoutSelection = function (data) {
+  var fakeCommentId = data.comment.commentId;
+  var newCommentId = shared.generateCommentId();
+  this.mapFakeComments[fakeCommentId] = newCommentId;
+  var originalCommentId = data.comment.originalCommentId;
+  this.mapOriginalCommentsId[originalCommentId] = newCommentId;
+  data.comment.commentId = newCommentId;
+
+  // we don't need to wait for a commentId to save because the comment is already saved
+  this.socket.emit('addComment', data, function (){});
+
+  var commentId = data.comment.commentId;
+  var comment = data.comment;
+  this.setComment(commentId, comment);
+  this.shouldCollectComment = true;
+}
+
+ ep_comments.prototype.getMapfakeComments = function(){
+   return this.mapFakeComments;
+ }
+
+ // commentReplyData = {c-reply-123:{commentReplyData1}, c-reply-234:{commentReplyData1}, ...}
+ ep_comments.prototype.saveCommentReplies = function(padId, commentReplyData){
+   var self = this;
+   var data = self.buildCommentReplies(commentReplyData);
+   self.socket.emit('bulkAddCommentReplies', padId, data, function (replies){
+    _.each(replies,function(reply){
+      self.setCommentReply(reply);
+    });
+    self.shouldCollectComment = true; // force collect the comment replies saved
+   });
+ }
+
+ ep_comments.prototype.buildCommentReplies = function(repliesData){
+  var self = this;
+  var replies = _.map(repliesData, function(replyData){
+    return self.buildCommentReply(replyData);
+  });
+  return replies;
+ }
+
+ // take a replyData and add more fields necessary. E.g. 'padId'
+ ep_comments.prototype.buildCommentReply = function(replyData){
+   var data = this.getCommentData();
```

Comment:
do we need more than just the pad id? If we only need pad id, we don't need to call `getCommentData()` -- it gets much more info than needed. Simply use `data.padId = this.padId;`